Factory/HttpFoundationFactory.php

```diff
@@ -50,7 +60,7 @@ public function createRequest(ServerRequestInterface $psrRequest)
         $parsedBody = $psrRequest->getParsedBody();
         $parsedBody = is_array($parsedBody) ? $parsedBody : array();
 
-        $request = new Request(
+        $request = new $this->requestClass(
```

Comment:
Can we be sure that the constructor parameters match for all subclasses?

Reply:
no we cannot. PHP does not make such guarantee.

Reply:
@jdreesen If `Request::create()` can be used instead of `new Request()`, we may have the same result with a simple change of one line.